atlasdb-conjure/src/main/java/com/palantir/atlasdb/http/v2/ConjureJavaRuntimeTargetFactory.java

```diff
@@ -103,14 +107,32 @@ private ConjureJavaRuntimeTargetFactory() {
             ClientOptions clientOptions) {
         ClientConfiguration clientConfiguration = clientOptions.serverListToClient(serverListConfig);
 
-        T client = JaxRsClient.create(
+        T client = wrapConjureClientWithOkHttpTimeoutHandling(type, () -> JaxRsClient.create(
                 type,
                 addAtlasDbRemotingAgent(parameters.userAgent()),
                 HOST_METRICS_REGISTRY,
-                clientConfiguration);
+                clientConfiguration));
         return decorateFailoverProxy(type, () -> client);
     }
 
+    private static <T> T wrapConjureClientWithOkHttpTimeoutHandling(Class<T> iface, Supplier<T> clientFactory) {
+        return ReplaceIfExceptionMatchingProxy.newProxyInstance(
+                iface,
+                Suppliers.memoizeWithExpiration(clientFactory::get, 20, TimeUnit.MINUTES),
+                ConjureJavaRuntimeTargetFactory::isPossiblyOkHttpTimeoutBug);
```

Comment:
What this does:
- The client is replaced not more than once every 20 minutes.
- In the event we get a SocketTimeoutException or something with that somewhere in the causal chain, we replace the client (subject to the above limitation).